PurchasesCoreSwift/Identity/IdentityManager.swift

```diff
@@ -129,29 +123,24 @@ extension IdentityManager {
 
     @objc(createAliasForAppUserID:completion:)
     public func createAlias(appUserID alias: String, completion: @escaping (Error?) -> Void) {
-        guard let currentAppUserID = maybeCurrentAppUserID else {
-            Logger.warn(Strings.identity.creating_alias_failed_null_currentappuserid)
+        guard !currentAppUserID.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty,
+              !alias.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
             completion(ErrorUtils.missingAppUserIDError())
```

Comment:
should we `fatalError` here? or is an empty string less "bad" than a nil one?

Reply:
This is an interesting scenario:
- `currentAppUserID` is non-nil, but somehow it's empty. 
- developer called `createAlias` with an empty alias appUserID

Seems to me that: 
- if the current one is empty, that's bad, but it's recoverable as long as the alias is valid. 
- if the current one is empty and we don't allow for the alias to go through, there's no way to recover and the app will continue to function with an empty appUserID

So it feels like two different cases here that need to be handled independently, one of which allows for the alias to go through and the other one would prevent the backend call. 

Thoughts?

Reply:
Makes sense to me. Sounds like the only way to get an empty `currentAppUserId` is if the empty alias is allowed to the backend, though I suppose with old versions in the wild we still need the check.

Reply:
It looks like I can switch this to just ensure `createAlias` is called with a non-empty alias and that satisfies both scenarios. That is, removing this check:
`!currentAppUserID.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty`
and leaving just 
```swift
        guard !alias.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
            completion(ErrorUtils.missingAppUserIDError())
            return
        }
```

Reply:
(See latest commit) https://github.com/RevenueCat/purchases-ios/pull/743/commits/57cdc91755e14059274bc6345c18415a658044a9